openmc/mesh.py

```diff
@@ -257,7 +257,9 @@ def centroids(self):
 
         """
         ndim = self.n_dimension
-        vertices = self.vertices
+        # this line ensures that the vertices aren't adjusted by the origin or
+        # converted to the Cartesian system for cylindrical and spherical meshes
+        vertices = StructuredMesh.vertices.fget(self)
```

Comment:
```suggestion
        vertices = StructuredMesh._generate_vertices(*self._grids)
```
Sorry for not catching this bug in the earlier PR, but I think I prefer using the existing private methods/attributes more explicitly here.